### PR TITLE
Add eslint kit and use it on `main.js` and `lib/*.js` files

### PR DESCRIPTION
**Pull request by BigBlueHat**
Related to #46 

I tried to build up from a consistent style guide https://github.com/airbnb/javascript, but tweaked it to match reality. 😃 

Tests still pass too! 😁 

Hope this helps,
🎩

### CODE REVIEW

**Comment by iherman** [comment]
Well...

- Some of the changes are fine (a more systematic use of `const` instead of `let`, for example, or creation of a new object for `throw`) and there are some inconsistencies in using spaces in function arguments that were clearly my mistake that the script found. Thanks for those.
- For me, many of these stylistic changes have a flair of a 'non invented here syndrome' (not you personally, @BigBlueHat, but `eslint`). I see no rational reason for those changes, like the extra space between an `if` statement and the following brackets, or the usage of `'` instead of `"` for strings. I must admit some of those changes are also hurting my eyes, like cutting conditional statement into several lines. 

However.  If @tripu also agrees that this is the house style for W3C, I can yield; I do not care about this all that much. (Except for keeping to the 4 space indentation. I would be genuinely concerned about not grasping the right structure of the code with 2 spaces. I think the usage of 2 spaces is actually dangerous. But we have an agreement on that.)

That being said: I do not yet trust my own tests. @BigBlueHat have you used this version for some of the minutes (JSON, WP) of the past weeks? I would prefer to be sure about those before merging this... 

---

(B.t.w., the style I am using stuck to me since my early C/C++ programming days when there was no Javascript yet. I inherited some style of use back in those days; hence my reference to a NIH syndrome. I cannot precisely trace it back because it was a while ago, but it was a time when there were several "clean" styles were in use and people/groups could choose among those instead of having only one... But I am ready to acknowledge that I am an old fart who does not understand this new age,).

File: .editorconfig [hunk 1]
```diff
@@ -7,3 +7,6 @@ end_of_line = lf
 charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
+
+[*.yml]
+indent_size = 2
```
Comment (tripu) [inline .editorconfig]: Why this exception for YML files?
Shouldn't we be consistent with what we agreed (4 spaces)?

File: README.md [hunk 2]
```diff
@@ -168,3 +168,15 @@ your platform, you can use it like so:
...
 npm test -- --growl
 ```
+
+## Linting
+
+We like clean code, so we've introduced tools to help our shared consistency.
+You can use `eslint` in the context of this project by running:
+
+```sh
+npm lint -- main.js
```
Comment (tripu) [inline README.md]: `npm run lint -- main.js`
(`run` is missing)

**Comment by tripu** [comment]
Once merged, I think we can close #46.

**Comment by iherman** [comment]
I cannot make those changes in the version of @BigBlueHat ; I will merge this and do it in the merged version (just to close this).

**Comment by BigBlueHat** [comment]
Thanks for merging this, @iherman and for your follow up work in #51 @tripu. If there are more eslint related tweaks we want/need made, I'm happy to explore them.

Most of those "NIH" things are highly preferential, so if there's a desire to change them back, we can. There are often alternate settings or config options to support most stylistic differences.

Cheers!
🎩

**Pull request merged**